src/fprime/util/cli.py

```diff
@@ -179,4 +225,54 @@ def add_special_parsers(
         action="store_true",
         help="Tells the new command to generate a port",
     )
-    return {"hash-to-file": hash_to_file, "info": print_info, "new": template}
+
+    # Code formatting with clang-format
+    format_parser = subparsers.add_parser(
+        "format",
+        help=help_text.short("format"),
+        description=help_text.long("format"),
+        parents=[common],
+        add_help=True,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        conflict_handler="resolve",
+    ).add_argument_group("format utility arguments")
+    format_parser.add_argument(
+        "-x", "--no-backup", action="store_true", help="Disable backups"
+    )
+    format_parser.add_argument(
+        "-q", "--quiet", action="store_true", help="Disable clang-format verbose mode"
+    )
+    format_parser.add_argument(
+        "--force",
+        action="store_true",
+        help="Force all listed files to be passed to clang-format (no file extension check)",
+    )
+    format_parser.add_argument(
+        "--whitelist",
+        action="append",
+        default=[],
+        help="Add a file extension to the allowed set",
+    )
+    format_parser.add_argument(
+        "-f",
```

Comment:
Should we make these positional inputs?  It would be more intuitive to users to not need to specify -f just to format.

Reply:
I agree that it would be more intuitive. However because the common parser has the `platform` optional positional argument, I believe this is not possible, since the parser can't tell whether the first positional is `platform` or `files`. Unless I missed an option to influence the parsing or remove the platform argument from the child parser.